framework/src/utils/MooseUtils.C

```diff
@@ -598,3 +560,70 @@ toUpper(const std::string & name)
 }
 
 } // MooseUtils namespace
+
+std::string
+getLatestCheckpointFileHelper(const std::list<std::string> & checkpoint_files,
+                              const std::vector<std::string> extensions,
+                              bool keep_extension)
+{
+  // Create storage for newest restart files
+  // Note that these might have the same modification time if the simulation was fast.
+  // In that case we're going to save all of the "newest" files and sort it out momentarily
+  time_t newest_time = 0;
+  std::list<std::string> newest_restart_files;
+
+  // Loop through all possible files and store the newest
+  for (const auto & cp_file : checkpoint_files)
+  {
+    if (find_if(extensions.begin(), extensions.end(), [cp_file](std::string ext) {
```

Comment:
OK, I'm not an expert with lambdas, but should the `ext` arg be a const reference here?